FlowCryptAppTests/Functionallity/Services/Organisational Rules Service/ClientConfigurationServiceTest.swift

```diff
@@ -0,0 +1,167 @@
+//
+//  ClientConfigurationServiceTest.swift
+//  FlowCryptAppTests
+//
+//  Created by Anton Kharchevskyi on 10.09.2021.
+//  Copyright © 2021 FlowCrypt Limited. All rights reserved.
+//
+
+import XCTest
+@testable import FlowCrypt
+
+// check if Email Key Manager should be used test and other client configuration is consistent
+class ClientConfigurationServiceTest: XCTestCase {
+
+    var sut: ClientConfigurationService!
+    var organisationalRulesService = OrganisationalRulesServiceMock()
+
+    override func setUp() {
+        super.setUp()
+
+        sut = ClientConfigurationService(organisationalRulesService: organisationalRulesService)
+    }
+
+    func testCheckDoesNotUseEKM() {
+        // EKM should not be used if keyManagerUrl is nil
+        organisationalRulesService.clientConfiguration = ClientConfiguration(keyManagerUrl: nil)
+        XCTAssert(sut.checkShouldUseEKM() == .doesNotUseEKM)
+
+        // EKM should not be used if keyManagerUrl is nil
+        organisationalRulesService.clientConfiguration = ClientConfiguration(
+            flags: [],
+            keyManagerUrl: nil
+        )
+
+        organisationalRulesService.clientConfiguration = ClientConfiguration(
+            flags: [.forbidStoringPassphrase],
+            keyManagerUrl: nil
+        )
+        XCTAssert(sut.checkShouldUseEKM() == .doesNotUseEKM)
+    }
+
+    func testShouldUseEKM() {
+        organisationalRulesService.clientConfiguration = ClientConfiguration(
+            flags: [
+                .privateKeyAutoimportOrAutogen,
+                .forbidStoringPassphrase
+            ],
+            keyManagerUrl: "https://ekm.example.com"
+        )
+
+        XCTAssert(sut.checkShouldUseEKM() == .usesEKM)
+    }
```

Comment:
As per this: 

![image](https://user-images.githubusercontent.com/6306961/134242846-f288cdac-4c8e-479c-b6e0-e3b643268dd8.png)

taken from here: https://github.com/FlowCrypt/flowcrypt-ios/issues/279

These flags as set up should produce an error, because the flags are missing `NO_PRV_CREATE` which translates to `.forbidCreatingPrivateKey`. I suppose we don't have this flag implemented in the app, so I'll make a separate issue for it.